Error raising error in calc_packaging

I'm trying to use comtypes to interface with a vendor provided COM interface. When I run it, I get the following error message:

```
(wyatt) PS C:\wyatt-control> python .\test_api.py
Traceback (most recent call last):
  File "C:\wyatt-control\test_api.py", line 11, in <module>
    astra_com = comtypes.client.CreateObject("WTC.Astra8.Application.1")
  File "C:\Users\biocat\miniconda3\envs\wyatt\lib\site-packages\comtypes\client\_create.py", line 122, in CreateObject
    return _manage(obj, clsid, interface=interface)
  File "C:\Users\biocat\miniconda3\envs\wyatt\lib\site-packages\comtypes\client\_managing.py", line 113, in _manage
    obj = GetBestInterface(obj)
  File "C:\Users\biocat\miniconda3\envs\wyatt\lib\site-packages\comtypes\client\_managing.py", line 87, in GetBestInterface
    mod = GetModule(tlib)
  File "C:\Users\biocat\miniconda3\envs\wyatt\lib\site-packages\comtypes\client\_generate.py", line 127, in GetModule
    return ModuleGenerator(tlib, pathname).generate()
  File "C:\Users\biocat\miniconda3\envs\wyatt\lib\site-packages\comtypes\client\_generate.py", line 232, in generate
    wrp_code = codegen.generate_wrapper_code(items, filename=self.pathname)
  File "C:\Users\biocat\miniconda3\envs\wyatt\lib\site-packages\comtypes\tools\codegenerator\codegenerator.py", line 202, in generate_wrapper_code
    self.generate_all(items)
  File "C:\Users\biocat\miniconda3\envs\wyatt\lib\site-packages\comtypes\tools\codegenerator\codegenerator.py", line 110, in generate_all
    self.generate(item)
  File "C:\Users\biocat\miniconda3\envs\wyatt\lib\site-packages\comtypes\tools\codegenerator\codegenerator.py", line 106, in generate
    mth(item)
  File "C:\Users\biocat\miniconda3\envs\wyatt\lib\site-packages\comtypes\tools\codegenerator\codegenerator.py", line 370, in Structure
    self.generate(struct.get_body())
  File "C:\Users\biocat\miniconda3\envs\wyatt\lib\site-packages\comtypes\tools\codegenerator\codegenerator.py", line 106, in generate
    mth(item)
  File "C:\Users\biocat\miniconda3\envs\wyatt\lib\site-packages\comtypes\tools\codegenerator\codegenerator.py", line 396, in StructureBody
    pack = packing.calc_packing(body.struct, fields)
  File "C:\Users\biocat\miniconda3\envs\wyatt\lib\site-packages\comtypes\tools\codegenerator\packing.py", line 58, in calc_packing
    raise PackingError(f"PACKING FAILED: {details}")
UnboundLocalError: local variable 'details' referenced before assignment
```

Looking at the calc_packing code, the error arises because details is being used to catch the error, and in python 3 such variables are deleted after the except block finishes. I think it could be fixed by changing the except as:
```
        except PackingError as e:
            details = e
            continue
```

Here's the error:

```
2026-05-11 10:17:31,344 - comtypes.client._generate - MainThread - INFO - Could not import comtypes.gen._368D43B2_3A78_4EB0_93D1_5339084555E2_0_1_0: 40
2026-05-11 10:17:31,347 - comtypes.client._generate - MainThread - INFO - # Generating comtypes.gen._368D43B2_3A78_4EB0_93D1_5339084555E2_0_1_0
C:\Users\biocat\miniconda3\envs\wyatt\lib\site-packages\comtypes\tools\codegenerator\codegenerator.py:403: UserWarning: Structure UvDeviceDetails: PACKING FAILED: field deviceModel offset (32/64)
  warnings.warn(message, UserWarning)
C:\Users\biocat\miniconda3\envs\wyatt\lib\site-packages\comtypes\tools\codegenerator\codegenerator.py:403: UserWarning: Structure BaselineDetails: PACKING FAILED: field type offset (32/64)
  warnings.warn(message, UserWarning)
C:\Users\biocat\miniconda3\envs\wyatt\lib\site-packages\comtypes\tools\codegenerator\codegenerator.py:403: UserWarning: Structure LogonResult: PACKING FAILED: field errorMessage offset (32/64)
  warnings.warn(message, UserWarning)
C:\Users\biocat\miniconda3\envs\wyatt\lib\site-packages\comtypes\tools\codegenerator\codegenerator.py:403: UserWarning: Structure SampleInfo: PACKING FAILED: field description offset (32/64)
  warnings.warn(message, UserWarning)
2026-05-11 10:17:31,424 - comtypes.client._generate - MainThread - INFO - # Generating comtypes.gen.AstraLib
Traceback (most recent call last):
  File "C:\wyatt-control\wyatt_control_com.py", line 138, in <module>
    wc = WyattControl()
  File "C:\wyatt-control\wyatt_control_com.py", line 79, in __init__
    self.astra_com = comtypes.client.CreateObject("WTC.ASTRA8.Application.1")
  File "C:\Users\biocat\miniconda3\envs\wyatt\lib\site-packages\comtypes\client\_create.py", line 122, in CreateObject
    return _manage(obj, clsid, interface=interface)
  File "C:\Users\biocat\miniconda3\envs\wyatt\lib\site-packages\comtypes\client\_managing.py", line 113, in _manage
    obj = GetBestInterface(obj)
  File "C:\Users\biocat\miniconda3\envs\wyatt\lib\site-packages\comtypes\client\_managing.py", line 87, in GetBestInterface
    mod = GetModule(tlib)
  File "C:\Users\biocat\miniconda3\envs\wyatt\lib\site-packages\comtypes\client\_generate.py", line 127, in GetModule
    return ModuleGenerator(tlib, pathname).generate()
  File "C:\Users\biocat\miniconda3\envs\wyatt\lib\site-packages\comtypes\client\_generate.py", line 240, in generate
    return [_create_module(name, code) for (name, code) in codebases][-1]
  File "C:\Users\biocat\miniconda3\envs\wyatt\lib\site-packages\comtypes\client\_generate.py", line 240, in <listcomp>
    return [_create_module(name, code) for (name, code) in codebases][-1]
  File "C:\Users\biocat\miniconda3\envs\wyatt\lib\site-packages\comtypes\client\_generate.py", line 212, in _create_module
    return _my_import(modulename)
  File "C:\Users\biocat\miniconda3\envs\wyatt\lib\site-packages\comtypes\client\_generate.py", line 26, in _my_import
    return importlib.import_module(fullname)
  File "C:\Users\biocat\miniconda3\envs\wyatt\lib\importlib\__init__.py", line 126, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
  File "<frozen importlib._bootstrap>", line 1050, in _gcd_import
  File "<frozen importlib._bootstrap>", line 1027, in _find_and_load
  File "<frozen importlib._bootstrap>", line 1006, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 688, in _load_unlocked
  File "<frozen importlib._bootstrap_external>", line 883, in exec_module
  File "<frozen importlib._bootstrap>", line 241, in _call_with_frames_removed
  File "C:\Users\biocat\miniconda3\envs\wyatt\lib\site-packages\comtypes\gen\_368D43B2_3A78_4EB0_93D1_5339084555E2_0_1_0.py", line 159, in <module>
    assert sizeof(UvDeviceDetails) == 72, sizeof(UvDeviceDetails)
AssertionError: 44
```

Files:
[_368D43B2_3A78_4EB0_93D1_5339084555E2_0_1_0.py](https://github.com/user-attachments/files/27601453/_368D43B2_3A78_4EB0_93D1_5339084555E2_0_1_0.py)
[_420B2830_E718_11CF_893D_00A0C9054228_0_1_0.py](https://github.com/user-attachments/files/27601454/_420B2830_E718_11CF_893D_00A0C9054228_0_1_0.py)
[_00020430_0000_0000_C000_000000000046_0_2_0.py](https://github.com/user-attachments/files/27601455/_00020430_0000_0000_C000_000000000046_0_2_0.py)

--- a/comtypes/test/test_packing.py
+++ b/comtypes/test/test_packing.py
@@ -1,0 +1,49 @@
+import unittest
+
+from comtypes.tools import typedesc
+from comtypes.tools.codegenerator.packing import PackingError, calc_packing
+
+
+# Sizes, alignments and offsets are expressed in bits, matching the values
+# produced by ``comtypes.tools.tlbparser`` (e.g. a 32-bit ``int``).
+def _make_field(name, size, align, offset):
+    typ = typedesc.FundamentalType("int", size, align)
+    return typedesc.Field(name, typ, None, offset)
+
+
+class CalcPackingTest(unittest.TestCase):
+    def test_returns_none_for_default_packing(self):
+        # A naturally laid out single-field struct needs no explicit packing.
+        field = _make_field("a", 32, 32, 0)
+        struct = typedesc.Structure(
+            "Good", align=32, members=[field], bases=[], size=32
+        )
+        self.assertIsNone(calc_packing(struct, [field]))
+
+    def test_incomplete_struct_returns_none(self):
+        field = _make_field("a", 32, 32, 0)
+        struct = typedesc.Structure(
+            "Incomplete", align=32, members=[field], bases=[], size=None
+        )
+        self.assertIsNone(calc_packing(struct, [field]))
+
+    def test_raises_packing_error_with_details_when_layout_is_inconsistent(self):
+        # The declared field offset does not match the computed layout, so every
+        # packing attempt raises ``PackingError`` and ``calc_packing`` must
+        # re-raise with the underlying reason.
+        #
+        # Regression test for gh-937: the final ``raise`` referenced the
+        # ``except ... as`` target after it had been cleared, raising
+        # ``UnboundLocalError`` and masking the real ``PackingError``.
+        field = _make_field("x", 32, 32, 64)
+        struct = typedesc.Structure("Bad", align=32, members=[field], bases=[], size=96)
+        with self.assertRaises(PackingError) as ctx:
+            calc_packing(struct, [field])
+        message = str(ctx.exception)
+        self.assertIn("PACKING FAILED", message)
+        # The original failure reason must be preserved, not lost.
+        self.assertIn("field x offset", message)
+
+
+if __name__ == "__main__":
+    unittest.main()

--- a/comtypes/tools/codegenerator/packing.py
+++ b/comtypes/tools/codegenerator/packing.py
@@ -44,10 +44,15 @@ def _calc_packing(struct, fields, pack, isStruct):
 def calc_packing(struct, fields):
     # try several packings, starting with unspecified packing
     isStruct = isinstance(struct, typedesc.Structure)
+    last_error = None
     for pack in [None, 16 * 8, 8 * 8, 4 * 8, 2 * 8, 1 * 8]:
         try:
             _calc_packing(struct, fields, pack, isStruct)
         except PackingError as details:
+            # The exception target is cleared when the ``except`` block ends
+            # (see https://docs.python.org/3/reference/compound_stmts.html#except-clause),
+            # so keep a reference for the final error message below.
+            last_error = details
             continue
         else:
             if pack is None:
@@ -55,7 +60,7 @@ def calc_packing(struct, fields):
 
             return int(pack / 8)
 
-    raise PackingError(f"PACKING FAILED: {details}")
+    raise PackingError(f"PACKING FAILED: {last_error}")
 
 
 class PackingError(Exception):

